lib/mix/lib/mix/dep/loader.ex

```diff
@@ -320,6 +322,13 @@ defmodule Mix.Dep.Loader do
     end
   end
 
+  defp incompatible_requirement?(%Mix.Dep{opts: opts}) do
+    case opts[:lock] do
+      {scm, _, _, _} -> is_nil(opts[scm])
```

Comment:
Oh, I did not check the milestones.
I'll try to look at something a bit less urgent then as I am not really familiar with the project yet.
Thanks!